packages/plugin-conversation/src/encryption-transforms.js

```diff
@@ -0,0 +1,272 @@
+          }
+        });
+      }
+      // If we made it this far and still don't have an encryption key, assume
+      // this is a conversation that is not encrypted and we're performing an
+      // action that should not encrypt it (e.g. `leave`)
+      else {
+        Reflect.deleteProperty(activity, `kmsMessage`);
+      }
+    }
+  },
+
+  encryptAddActivity: {
+    direction: `outbound`,
+    alias: `encryptVerbActivity`
+  },
+
+  encryptCreateActivity: {
+    direction: `outbound`,
+    alias: `encryptVerbActivity`
+  },
+
+  encryptPostActivity: {
+    direction: `outbound`,
+    alias: `encryptVerbActivity`
+  },
+
+  encryptShareActivity: {
+    direction: `outbound`,
+    alias: `encryptVerbActivity`
+  },
+
+  encryptUpdateActivity: {
+    direction: `outbound`,
+    fn(ctx, key, activity) {
+      return ctx.transform(`encryptVerbActivity`, key, activity)
+        .then(() => {
+          key = key || activity[KEY];
+          activity.encryptionKeyUrl = key.uri || key;
+        });
+    }
+  },
+
+  encryptUpdateKeyActivity: {
+    direction: `outbound`,
+    alias: `encryptVerbActivity`
+  },
+
+  encryptComment(ctx, key, comment) {
+    return Promise.all([
+      ctx.transform(`encryptPropDisplayName`, key, comment),
+      ctx.transform(`encryptPropContent`, key, comment)
+    ]);
+  },
+
+  encryptContent(ctx, key, content) {
+    const promises = content.files.items.map((item) => ctx.transform(`encryptObject`, key, item));
+    promises.push(ctx.transform(`encryptPropContent`, key, content));
+    promises.push(ctx.transform(`encryptPropDisplayName`, key, content));
+    return Promise.all(promises);
+  },
+
+  encryptFile(ctx, key, file) {
+    if (file.image && !file.image.scr) {
+      return Promise.reject(new Error(`\`file.image\` must have an \`scr\``));
+    }
+
+    return Promise.all([
+      ctx.transform(`encryptPropScr`, key, file),
+      ctx.transform(`encryptPropDisplayName`, key, file),
+      ctx.transform(`encryptPropContent`, key, file),
+      file.image && ctx.transform(`encryptPropScr`, key, file.image)
+    ]);
+  },
+
+  encryptImageURI(ctx, key, imageURI) {
+    return ctx.transform(`encryptPropLocation`, key, imageURI);
+  },
+
+  encryptPropContent: encryptTextProp(`content`),
+
+  encryptPropDisplayName: encryptTextProp(`displayName`),
+
+  encryptPropLocation: encryptTextProp(`location`),
+
+  encryptPropScr(ctx, key, object) {
+    if (!object.scr) {
+      return Promise.resolve();
+    }
+
+    return ctx.spark.encryption.encryptScr(key, object.scr)
+      .then((scr) => {
+        object.scr = scr;
+      });
+  },
+
+  encryptTextProp(ctx, name, key, object) {
+    if (!object[name]) {
+      return Promise.resolve();
+    }
+
+    return ctx.spark.encryption.encryptText(key.uri || key, object[name])
+      .then((ciphertext) => {
+        object[name] = ciphertext;
+
+        return ctx.spark.encryption.decryptText(key.uri || key, object[name]);
```

Comment:
I think that's debugging code i didn't remove yet. i was getting weird errors on the response, so as a sanity check i was making sure crypto worked.